src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -216,13 +216,6 @@ StorageObjectStorage::StorageObjectStorage(
            context);
     }
 
-    if (hive_partition_columns_to_read_from_file_path.size() == columns.size())
```

Comment:
Is this check not required for `partition_strategy_type=hive` case?
Before check covered that case, now checking is made only inside `extractPartitionColumnsFromPathAndEnrichStorageColumns`.